File: app/src/main/java/torille/fi/lurkforreddit/subreddits/SubredditsContract.java
package torille.fi.lurkforreddit.subreddits;

import java.util.List;

import torille.fi.lurkforreddit.BasePresenter;
import torille.fi.lurkforreddit.BaseView;
import torille.fi.lurkforreddit.data.models.view.Subreddit;

/**
 * Created by eva on 2/8/17.
 */

public interface SubredditsContract {

    interface View extends BaseView {

        void setProgressIndicator(boolean active);

        void showSubreddits(List<Subreddit> subreddits);

        void loadSelectedSubreddit(Subreddit subreddit);

        void onError(String errorText);
    }

    interface Presenter<T extends BaseView> extends BasePresenter<View> {

        void loadSubreddits(boolean b);

        void openSubreddit(Subreddit subreddit);

    }

}

File: app/src/main/java/torille/fi/lurkforreddit/subreddits/SubredditsContract.kt
package torille.fi.lurkforreddit.subreddits

import torille.fi.lurkforreddit.BasePresenter
import torille.fi.lurkforreddit.BaseView
import torille.fi.lurkforreddit.data.models.view.Subreddit

/**
 * Binds view and presenter
 */

interface SubredditsContract {

    interface View : BaseView<Presenter> {

        fun setProgressIndicator(active: Boolean)

        fun showSubreddits(subreddits: List<Subreddit>)

        fun loadSelectedSubreddit(subreddit: Subreddit)

        fun onError(errorText: String)

    }

    interface Presenter : BasePresenter<View> {

        fun loadSubreddits(forcedUpdate: Boolean)

        fun openSubreddit(subreddit: Subreddit)

    }

}
